neps/space/neps_spaces/parameters.py

```diff
@@ -0,0 +1,1653 @@
+        """
+        if isinstance(self._kwargs, Resolvable):
+            raise ValueError(
+                f"Operation kwargs contain unresolved Resolvable: {self._kwargs!r}. "
+                "The operation needs to be resolved before accessing kwargs as a mapping."
+            )
+        return self._kwargs
+
+    def get_attrs(self) -> Mapping[str, Any]:
+        """Get the attributes of the operation as a mapping.
+        This method collects all attributes of the operation class and instance,
+        excluding private attributes and methods, and returns them as a dictionary.
+
+        Returns:
+            A mapping of attribute names to their values.
+
+        """
+        return {k.lstrip("_"): v for k, v in vars(self).items()}
+
+    def from_attrs(self, attrs: Mapping[str, Any]) -> Operation:
+        """Create a new Operation instance from the given attributes.
+
+        Args:
+            attrs: A mapping of attribute names to their values.
+
+        Returns:
+            A new Operation instance with the specified attributes.
+
+        Raises:
+            ValueError: If the attributes do not match the operation's expected structure.
+
+        """
+        return type(self)(**attrs)
+
+
+# TODO: [lum] For tuples, lists and dicts,
+#  should we make the behavior similar to other resolvables,
+#  in that they will be cached and then we also need to use Resampled for them?
+
+
+class Resampled(Resolvable):
+    """A class representing a resampling operation in a NePS space.
+
+    Attributes:
+        source: The source of the resampling, which can be a resolvable object or a
+            string.
+    """
+
+    def __init__(self, source: Resolvable | str):
+        """Initialize the Resampled object with a source.
+
+        Args:
+            source: The source of the resampling, can be a resolvable object or a string.
+        """
+        self._source = source
+
+    def __str__(self) -> str:
+        return f"Resampled({self._source!s})"
+
+    @property
+    def source(self) -> Resolvable | str:
+        """Get the source of the resampling.
+
+        Returns:
+            The source of the resampling, which can be a resolvable object or a string
+
+        """
+        return self._source
+
+    @property
+    def is_resampling_by_name(self) -> bool:
+        """Check if the resampling is by name.
+
+        Returns:
+            True if the source is a string, indicating a resampling by name,
+            False if the source is a resolvable object.
+
+        """
+        return isinstance(self._source, str)
+
+    def get_attrs(self) -> Mapping[str, Any]:
+        """Get the attributes of the resampling source as a mapping.
+
+        Returns:
+          A mapping of attribute names to their values.
+
+        Raises:
+          ValueError: If the resampling is by name or the source is not resolvable.
+
+        """
+        if self.is_resampling_by_name:
```

Comment:
DRY: the validation part for both `get_attrs` and `from_attrs` are the same, potentially we could extract a `_get_resolvable_source` function.